Cubical/Algebra/CommRing/Ideal.agda

```diff
@@ -0,0 +1,45 @@
+{-
+  This is mostly for convenience, when working with ideals
+  (which are defined for general rings) in a commutative ring.
+-}
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+module Cubical.Algebra.CommRing.Ideal where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Powerset
+
+open import Cubical.Algebra.CommRing.Base
+open import Cubical.Algebra.Ring.Ideal renaming (IdealsIn to IdealsInRing)
+open import Cubical.Algebra.RingSolver.ReflectionSolving hiding (∣)
+
+
+private
+  variable
+    ℓ : Level
+
+IdealsIn : (R : CommRing {ℓ}) → Type _
+IdealsIn R = IdealsInRing (CommRing→Ring R)
+
+module _ (Ring@(R , str) : CommRing {ℓ}) where
+  open CommRingStr str
+  makeIdeal : (I : R → hProp ℓ)
+              → (+-closed : {x y : R} → x ∈ I → y ∈ I → (x + y) ∈ I)
+              → (0r-closed : 0r ∈ I)
+              → (·-closedLeft : {x : R} → (r : R) → x ∈ I → r · x ∈ I)
+              → IdealsIn (R , str)
+  makeIdeal I +-closed 0r-closed ·-closedLeft = I ,
+    (record
+       { +-closed = +-closed
+       ; -closed = λ x∈I → subst (_∈ I)
+                             (- 1r · _ ≡⟨ step1 _ ⟩ - _ ∎ )
```

Comment:
Just write `step1 _` here? Doesn't seem like the `≡⟨_⟩` notation clarifies much here, given that the endpoints are mostly `_` and the type signature of `step1` is already written out below.

Reply:
Yes!